Sources/StreamChat/StateLayer/ChannelListState+Observer.swift

```diff
@@ -43,5 +95,32 @@ extension ChannelListState {
                 log.error("Failed to start the channel list observer for query: \(query)")
             }
         }
+        
+        // MARK: Linking and Unlinking Channels from the Query
+        
+        private func isBelongingToChannelListQuery(channel: ChatChannel) -> Bool {
+            if let filter = dynamicFilter {
+                return filter(channel)
+            }
+            // When auto-filtering is enabled the channel will appear or not automatically if the
+            // query matches the DB Predicate. So here we default to saying it always belong to the current query.
+            if clientConfig.isChannelAutomaticFilteringEnabled {
+                return true
+            }
+            return false
+        }
+        
+        private func linkChannelIfNeeded(_ channel: ChatChannel) async throws {
+            guard !channelListObserver.items.contains(where: { $0.cid == channel.cid }) else { return }
```

Comment:
Not so nice, since it forces the observer to initiate channels and therefore losing the benefit of being a LazyCachedMapCollection. But this is how it is currently in the ChannelListController as well.

Another option is to have a separate fetch request (`channelListFetchRequest`, limit 1, filtered by cid). WDYT? (I would create a small separate PR if we go for it)

Reply:
I'll create a follow-up PR for this after it is merged. Added a note for myself.